src/Microsoft.Framework.PackageManager/Packages/Workers/FileSystemPackages.cs

```diff
@@ -0,0 +1,429 @@
+            if (record.Contents == null)
+            {
+                record.Contents = new List<string>();
+            }
+            return record;
+        }
+
+        private string FirstTwoParts(string path)
+        {
+            var parts = path.Split(new[] { '/' }, 3);
+            return string.Join("/", parts.Take(2));
+        }
+        private string FirstPart(string path)
+        {
+            var parts = path.Split(new[] { '/' }, 2);
+            return parts.First();
+        }
+    }
+
+    /// <summary>
+    /// Summary description for FileSystemPackages
+    /// </summary>
+    public class FileSystemRepositoryPublisher : AbstractRepositoryPublisher
+    {
+        private readonly string _path;
+
+        public FileSystemRepositoryPublisher(string path)
+        {
+            _path = path;
+        }
+
+        public override Stream ReadArtifactStream(string path)
+        {
+            var filePath = Path.Combine(_path, path);
+            if (!File.Exists(filePath))
+            {
+                return null;
+            }
+            return new FileStream(
+                filePath,
+                FileMode.Open,
+                FileAccess.Read,
+                FileShare.Read | FileShare.Delete);
+        }
+
+        public override void WriteArtifactStream(string path, Stream content, bool createNew)
+        {
+            var combinedPath = Path.Combine(_path, path);
+            var combinedDirectory = Path.GetDirectoryName(combinedPath);
+
+            Directory.CreateDirectory(combinedDirectory);
+
+            using (var stream = new FileStream(
+                combinedPath,
+                createNew ? FileMode.CreateNew : FileMode.Create,
+                FileAccess.Write,
+                FileShare.Delete))
+            {
+                content.CopyTo(stream);
+            }
+        }
+
+        public override void RemoveArtifact(string path)
+        {
+            var combinedPath = Path.Combine(_path, path);
+            if (File.Exists(combinedPath))
+            {
+                File.Delete(combinedPath);
+            }
+        }
+
+        public override IEnumerable<string> EnumerateArtifacts(
+            Func<string, bool> folderPredicate,
+            Func<string, bool> artifactPredicate)
+        {
+            List<string> result = new List<string>();
+            EnumerateArtifactsRecursive("", folderPredicate, artifactPredicate, result);
+            return result;
+        }
+
+        void EnumerateArtifactsRecursive(
+            string subPath,
+            Func<string, bool> folderPredicate,
+            Func<string, bool> artifactPredicate,
+            List<string> result)
+        {
+            foreach (var name in Directory.EnumerateDirectories(Path.Combine(_path, subPath)))
+            {
+                var directoryName = Path.GetFileName(name);
+                var directoryPath = Path.Combine(subPath, directoryName)
+                    .Replace("\\", "/");
```

Comment:
The indexing system of `kpm packages commit` consistently uses `/` as path separators. So we'd better keep this replacement in case `folderPredicate` or `artifactPredicate` depends on `/` as path separators.